cogs/perpetuum_killboard_cog.py

```diff
@@ -101,52 +158,88 @@ async def display_new_kills(self):
+
+            # Embed - Attacker(s) | Discord has some limitations when it comes to Embeds
+            # More info: https://discordjs.guide/popular-topics/embeds.html#embed-limits
+
+            # Setup for Attacker Field(s) for Embed
+            atk_field_name = "⚔ Attackers"
+            atkcount = (str(len(kill['_embedded']['attackers'])))
+            if int(atkcount) <= 1:  # If there's only 1 attacker, change field name to singular.
+                atk_field_name = "⚔ Attacker"
+
+            atk_saved_list = ""  # Validated Attackers, saved to a list - These WILL exist in the embed
+            current_atk_str = ""  # Validated Attackers + New Attacker we want to add - Check if this exceeds limits
+            current_field_length = 0  # char-length of field we're currently manipulating
+            embed_length = len(kill_message_embed)  # Embed char length, MAX 6000
+            fields_built = len(kill_message_embed.fields)  # Keep track of total fields, MAX 25
+            inline_next_attacker_field = False
+
+            # Find Killing Blow and add to the list first
+            for a in kill['_embedded']['attackers']:
                 if a["hasKillingBlow"]:
-                    attack_text += " Killing Blow!"
-
-                attackers_text += attack_text + "\n"
-
-            kill_message = raw_kill_message.format(
-                kill['_embedded']['agent']['name'],
-                kill['_embedded']['zone']['name'],
-                kill['_embedded']['corporation']['name'],
-                kill['date'],
-                bot_name_lookup.get(kill['_embedded']['robot']['definition'], kill['_embedded']['robot']['definition']),
-                attackers_text)
-
+                    atk_saved_list = add_attacker_str(atk_saved_list, a)
+                    kill['_embedded']['attackers'].remove(a)  # Attacker has now been added, remove it from list
+                    # Edge case, if there was only 1 attacker in total, we want to post the message
+                    if len(kill['_embedded']['attackers']) == 0:
+                        build_attacker_field(atk_saved_list, kill_message_embed, atk_field_name)
+                    break
+
+            # Validate the rest of the attackers
+            for a in kill['_embedded']['attackers']:
+                current_atk_str = add_attacker_str(atk_saved_list, a)  # Add a new Attackers info to the Saved list
+                embed_length = len(kill_message_embed)  # Update current embed length
+                new_length = len(current_atk_str) + embed_length
+
+                if new_length >= 5800:  # Do we exceed the allowed Embed Length?
+                    # End early, build what we have and add 'too many attackers' field at the end of the embed
+                    build_attacker_field(atk_saved_list, kill_message_embed, atk_field_name)
+                    add_too_many_attackers(kill_message_embed)  # Costs 1 field + ~85 characters
```

Comment:
_L219 & L233:_
This is a coroutine call, python would prefer if we used `await` when calling this.
However, the library we are using to interact with Discord does not allow for Embeds to be awaited - If you try you get:
`TypeError: object Embed can't be used in 'await' expression`

I have not run into any issues by *not* using `await` here, but your code editor might scream at you :)